### PR TITLE
Update jaxrs-common to Instrumenter API

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/HandlerData.java [hunk 1]
```diff
@@ -1,106 +1,42 @@
...
-  public void updateSpanNames(
-      Context context, Span span, Span serverSpan, Class<?> target, Method method) {
-    Supplier<String> spanNameSupplier = getPathSpanNameSupplier(context, target, method);
-    if (serverSpan == null) {
-      updateSpanName(span, spanNameSupplier.get());
-    } else {
-      ServerSpanNaming.updateServerSpanName(
-          context, ServerSpanNaming.Source.CONTROLLER, spanNameSupplier);
-      updateSpanName(span, SpanNames.fromMethod(target, method));
-    }
```
Comment (trask) [inline instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/HandlerData.java:74]: I removed this special case of having a different name for the handler span name when there is no server span (now the handler span has the same name whether or not there is a server span, which is consistent with other instrumentations).

File: instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/HandlerData.java [hunk 2]
```diff
@@ -5,102 +5,43 @@
...
+    this.target = target;
+    this.method = method;
   }
 
-  private static void updateSpanName(Span span, String spanName) {
-    if (!spanName.isEmpty()) {
-      span.updateName(spanName);
-    }
+  String spanName() {
+    return SpanNames.fromMethod(target, method);
```
Comment (mateuszrzeszutek) [inline instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/HandlerData.java:36]: Why not `CodeSpanNameExtractor`?

File: instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/RequestContextHelper.java [hunk 3]
```diff
@@ -5,51 +5,46 @@
...
+
+    ServerSpanNaming.updateServerSpanName(
+        parentContext,
+        ServerSpanNaming.Source.CONTROLLER,
+        JaxrsServerSpanNaming.getServerSpanNameSupplier(parentContext, handlerData));
+
+    // if there's no current span or it's the same as the server (servlet) span we need to start
+    // a JAX-RS one
+    // in other case, DefaultRequestContextInstrumentation must have already run so it's enough
+    // to just update the names
```
Comment (mateuszrzeszutek) [inline instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/RequestContextHelper.java:38]: That's no longer applicable

File: instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-cxf-3.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/CxfRequestContextInstrumentation.java [hunk 4]
```diff
@@ -58,38 +59,46 @@ public void transform(TypeTransformer transformer) {
...
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void stopSpan(
+        @Local("otelHandlerData") HandlerData handlerData,
         @Local("otelContext") Context context,
         @Local("otelScope") Scope scope,
         @Advice.Thrown Throwable throwable) {
-      RequestContextHelper.closeSpanAndScope(context, scope, throwable);
+      if (scope == null) {
+        return;
+      }
```
Comment (mateuszrzeszutek) [inline instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-cxf-3.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/CxfRequestContextInstrumentation.java:100]: Missing `scope.close()`?

File: instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-resteasy-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/Resteasy30RequestContextInstrumentation.java [hunk 5]
```diff
@@ -36,30 +38,36 @@ protected String abortAdviceName() {
...
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void stopSpan(
+        @Local("otelHandlerData") HandlerData handlerData,
         @Local("otelContext") Context context,
         @Local("otelScope") Scope scope,
         @Advice.Thrown Throwable throwable) {
-      RequestContextHelper.closeSpanAndScope(context, scope, throwable);
+      if (scope == null) {
+        return;
+      }
```
Comment (mateuszrzeszutek) [inline instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-resteasy-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/Resteasy30RequestContextInstrumentation.java:69]: Missing `scope.close()`?

File: instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-resteasy-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/Resteasy31RequestContextInstrumentation.java [hunk 6]
```diff
@@ -36,30 +38,36 @@ protected String abortAdviceName() {
...
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void stopSpan(
+        @Local("otelHandlerData") HandlerData handlerData,
         @Local("otelContext") Context context,
         @Local("otelScope") Scope scope,
         @Advice.Thrown Throwable throwable) {
-      RequestContextHelper.closeSpanAndScope(context, scope, throwable);
+      if (scope == null) {
+        return;
+      }
```
Comment (mateuszrzeszutek) [inline instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-resteasy-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/Resteasy31RequestContextInstrumentation.java:69]: Missing `scope.close()`?

File: instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-cxf-3.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/CxfRequestContextInstrumentation.java (hunk 4 above)
Comment (trask) [reply instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-cxf-3.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/CxfRequestContextInstrumentation.java:100]: 😬

**Review by anuraaga — APPROVED** [review]

**Pull request merged**